apps/concept-catalog/hooks/change-requests.ts

```diff
@@ -66,9 +65,30 @@ export const useCreateChangeRequest = ({ catalogId }) => {
 };
 
 export const useUpdateChangeRequest = ({ catalogId, changeRequestId }) => {
+  const router = useRouter();
+  const queryClient = useQueryClient();
+
   return useMutation({
     mutationFn: async (changeRequest: ChangeRequestUpdateBody) => {
-      return Promise.reject('Not implemented');
+      if (!validOrganizationNumber(catalogId)) {
+        return Promise.reject('Invalid catalog id');
+      }
+      if (!validUUID(changeRequestId)) {
+        return Promise.reject('Invalid change request id');
+      }
+
+      const response = await fetch(`/api/change-requests/${catalogId}/updateChangeRequest/${changeRequestId}`, {
+        method: 'POST',
+        body: JSON.stringify(changeRequest),
+      });
+
+      if (response.status === 401) {
+        return Promise.reject('Unauthorized');
+      }
+    },
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: ['getChangeRequests', catalogId] });
+      router.reload();
```

Comment:
Du kan vurdere å gjøre router.reload() i onSuccess når man kaller mutate for å unngå at router gjør en reload for alle som bruker hooken. OnSuccess som er definert her blir fremdeles eksekvert men før onSuccess i mutate.

Reply:
I pagen, da?
```
changeRequestMutateHook.mutateAsync(changeRequestFromConcept)
    .then(() =>{
       router.reload()
    })
    .catch((error) => {
      alert('Failed to submit change request.');
    });
});
```

Reply:
Du kan bruke hook.mutate(id, { onSuccess: () => {}})